api/tests/unit/domain/services/encryption-service_test.js

```diff
@@ -1,47 +1,71 @@
-/* eslint-disable no-sync */
 const { sinon, expect } = require('../../../test-helper');
 
 const bcrypt = require('bcrypt');
 const encryptionService = require('../../../../lib/domain/services/encryption-service');
-
 const PasswordNotMatching = require('../../../../lib/domain/errors').PasswordNotMatching;
 
 describe('Unit | Service | Encryption', () => {
 
   describe('#check', () => {
 
-    it('should reject when passwords are not matching', () => {
-      // given
-      const encryptedPassword = bcrypt.hashSync('my-real-password', 1);
-      const password = 'my-expected-password';
-
-      // when
-      const promise = encryptionService.check(password, encryptedPassword);
-
-      // then
-      return promise
-        .then(() => {
-          sinon.assert.fail('Should not succeed');
-        })
-        .catch((err) => {
-          expect(err).to.be.an.instanceof(PasswordNotMatching);
+    describe('#when password and hash are matching', () => {
+      it('should resolve', () => {
+        // given
+        /* eslint-disable no-sync */
+        const passwordHashed = bcrypt.hashSync('my-real-password', 1);
+        const password = 'my-real-password';
+
+        // when
+        const promise = encryptionService.check(password, passwordHashed);
+
+        // then
+        return promise.catch((_) => {
+          sinon.assert.fail('Should not fail');
         });
+      });
     });
 
-    it('should resolve when passwords are matching', () => {
-      // given
-      const encryptedPassword = bcrypt.hashSync('my-real-password', 1);
-      const password = 'my-real-password';
+    describe('#when password and hash are not matching', () => {
+      it('should reject a PasswordNotMatching error ', () => {
+        // given
+        /* eslint-disable no-sync */
+        const passwordHashed = bcrypt.hashSync('my-real-password', 1);
+        const password = 'my-expected-password';
 
-      // when
-      const promise = encryptionService.check(password, encryptedPassword);
+        // when
+        const promise = encryptionService.check(password, passwordHashed);
 
-      // then
-      return promise.catch((_) => {
-        sinon.assert.fail('Should not fail');
+        // then
+        return promise
+          .then(() => {
+            sinon.assert.fail('Should not succeed');
+          })
+          .catch((err) => {
+            expect(err).to.be.an.instanceof(PasswordNotMatching);
+          });
       });
     });
 
+    describe('#when password is not supplied', () => {
+      it('should reject, except a PasswordNotMatching error ', () => {
```

Comment:
```suggestion
      it('should reject, but not a PasswordNotMatching error', () => {
```

Reply:
Corrigé

Reply:
Pas vraiment j'ai l'impression… 😄 
![image](https://user-images.githubusercontent.com/300823/103001400-3631a480-452d-11eb-9d90-c1f11fb10c6a.png)

Reply:
Next shoot.. 
![image](https://user-images.githubusercontent.com/56302715/103004266-e7d2d480-4531-11eb-89f1-bea47a1d9724.png)

Reply:
En anglais but sert aussi de "sauf" 
everything  but you => tout le monde sauf toi
Et je voulaire exprimer: il peut throw n'importe quoi sauf un `PasswordNotMatching`

Reply:
C'est pas faux pour le `but` ("should reject anything but…" marcherait), mais d'un autre côté le `except` n'était pas correct parce qu'il ne marche pas tout seul comme `sauf` : `except that …`, `except for …`. Et même en français, "doit rejeter, sauf une erreur PasswordNotMatching" serait bizarrement construit ?

Dommage sinon que l'espace en fin de chaîne ait résisté 😢

Reply:
On aurait pu solliciter @ebarbier (mais elle a mieux à faire) pour traduire
> Doit lever une erreur, mais pas PasswordNotMatching 

Uhmm.. pour l'esapce, je faisais confiance au linter :thinking: 
en 2021, rappelez-moi de rajouter une règle de lint
quoi de plus fourbe qu'un espace en fin de ligne ?